db/migrate/20230404064728_system_user_for_public_sections.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class SystemUserForPublicSections < ActiveRecord::Migration[7.0]
+  def up
+    DB.exec(<<~SQL, user_id: Discourse.system_user)
```

Comment:
I'm not sure exactly how `DB.exec` works but passing a `user` object doesn't seem like the right thing to do here. Also, I think it'll be better to pass in `-1` instead since `Discourse.system_user` is technically application code and is subjected to changes in the future.